### PR TITLE
fix(pie-chip): DSW-000 hover and active styles are applied for disabled state

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)

- Fixes an issue raised by design that the hover and active styles are applied when disabled is true.

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] If it is a component change, I have reviewed the Storybook preview
- [x] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-chip/src/chip.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @use '@justeattakeaway/pie-css/scss' as p;
 
 @mixin chip-interactive-states($bg-color, $mode: 'default') {
-    &:hover:not(:disabled) {
+    &:hover:not([disabled]) {
```
Comment (raoufswe) [inline packages/components/pie-chip/src/chip.scss:4]: the css`:disabled` selector is only applied if the element can be natively disabled

**Review by ashleynolan — APPROVED** [review]

File: .changeset/twenty-pianos-invent.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@justeattakeaway/pie-chip": minor
```
Comment (ashleynolan) [inline .changeset/twenty-pianos-invent.md:2]: If you wanted to, you could use patch instead of minor here (as it's a small fix). Not a biggie though 😆

**Review by dandel10n — APPROVED** [review]

File: packages/components/pie-chip/src/chip.scss (hunk 1 above)
Comment (jamieomaguire) [reply packages/components/pie-chip/src/chip.scss:4]: good spot!

**Review by jamieomaguire — APPROVED** [review]

File: .changeset/twenty-pianos-invent.md (hunk 2 above)
Comment (raoufswe) [reply .changeset/twenty-pianos-invent.md:2]: i will keep that in mind, thanks for flagging it 🥇

**Pull request merged**